components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -169,7 +169,14 @@ public Organization addOrganization(Organization organization) throws Organizati
             String tenantDomainName = ((TenantTypeOrganization) organization).getDomainName();
             createTenant(tenantDomainName, organization);
         }
-        getListener().postAddOrganization(organization);
+
+        try {
+            getListener().postAddOrganization(organization);
+        } catch (OrganizationManagementException e) {
+            // Rollback created organization.
+            deleteOrganization(organization.getId());
```

Comment:
We may have to rollback the changes done in the root organization.
Ex - When an organization is created, the b2b apps in the root org will get updated a property called isAppShared = true. if there at least one org. We may have to handle those scenarios also [This is least priority requirement]

Reply:
This may not required to be done as when the org is getting deleted, that mentioned attribute will be updated.

Reply:
yes this is handled at https://github.com/wso2-extensions/identity-organization-management/blob/92d0a9b48d431a85f70b2a44444745ef9599ee28/components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/listener/OrganizationCreationHandler.java#L214